test/decomposer/test_aba_decomposer.py

```diff
@@ -39,6 +39,6 @@ def test_full_sphere(decomposer_class: Callable[..., Decomposer]) -> None:
 
     for angle in angles:
         for axis in axis_list:
-            arbitrary_operation = BlochSphereRotation(qubit=0, axis=axis, angle=angle, phase=0.0)
+            arbitrary_operation = BlochSphereRotation(qubit=0, axis=axis, angle=angle, phase=0.1)
```

Comment:
Why is this change of phase from 0.0 to 0.1 needed?

Reply:
Yes, it tests out the new code fragment in check_gate_replacement

Reply:
A change from 0.0 to 0.1 tests out whether the `check_gate_replacement` can account for global phases, more specifically, if it is capable of identifying whether a matrix is an identity gate with a multiplied complex or negative factor. Before, this was not being picked up, thus [[i,0],[0,i]] was returning false to being an identity gate.

Reply:
Thanks  for the explanation. But I think that either we are mixing tests here, or we are not being exhaustive enough.

Can we:
1. Either have this `test_all_octants_of_bloch_sphere_rotation` focus on axis/angles combos, and then have another `test_all_phases_of_bloch_sphere_rotation` focus on phases?
2. Or, maybe better, add `phase` to the combinations here. I.e., we are testing a combination of axes and angles. Why not multiplying that combination to different phases?